api/api/serializers.py

```diff
@@ -51,16 +51,42 @@ class SpotSerializer(HALSerializer):
     rapport_document = serializers.FileField(use_url=True, required=False)
     design_document = serializers.FileField(use_url=True, required=False)
 
-    def create(self, validated_data):
-        # before creating the spot, we'll need to obtain the 'stadsdeel' based on the
-        # Point value.
-
-        point = validated_data.get('point')
+    def validate(self, attrs):
+        attrs = super().validate(attrs)
```

Comment:
Hier valideren we velden die afhankelijk zijn van andere velden, en niet los van elkaar gevalideerd kunnen worden:

- jaar_blackspotlijst is vereist als spot_type 'blackspot' of 'wegvak' is
- jaar_ongeval_quickscan is vereist als spot_type 'protocol_dodelijk' of 'protocol_ernstig' is
- stadsdeel is vereist, maar moet eerst in de bag opgezocht worden adhv Point